src/main/scala/org/dia/algorithms/mcc/GTGRunner.scala

```diff
@@ -324,18 +323,34 @@ class GTGRunner(val masterURL: String,
       minArea,
       nodeMinArea)
 
+    edgeListRDD.cache()
+    edgeListRDD.localCheckpoint()
+
     /**
      * Collect the edgeList and construct NodeMap
      */
     val MCCEdgeList = edgeListRDD.collect()
     val MCCNodeMap = createNodeMapFromEdgeList(MCCEdgeList, lat, lon)
 
+    val broadcastedNodeMap = sc.sparkContext.broadcast(MCCNodeMap)
+
     /**
      * Process the edge list. Collect and output edges and vertices
      */
     processEdges(MCCEdgeList, MCCNodeMap)
 
     /**
+     * Generate the netcdfs
+     */
+
+    val edgeTupleRDD = edgeListRDD.map({ x =>
+      (s"${x.srcNode.frameNum},${x.srcNode.cloudElemNum}",
+      s"${x.destNode.frameNum},${x.destNode.cloudElemNum}") })
+
```

Comment:
No, I kept the key in the nodeMap the same.

Both our PRs have edited the same file and it might create conflicts depending on which is merged first, lets do this - I have created a function in the MCCNode obj called hashKey(). Lets move to using that everywhere (in which case PR https://github.com/SciSpark/SciSpark/pull/154 would have to be merged first), thoughts ?